syntax = "proto3";



message CreateUserRequest {
    string username = 1;
    string email = 2;
    string password = 3;
    int32 phone_no = 4;

}

message CreateUserResult {
    int32 status = 1;
}

message CheckRequest {
    string email = 1;
    string password = 2;

}

message CheckResult {
    bool success = 1;
}

message UpdateUserRequest {
    string email = 1;
    string uname = 2;
    int32 uphone = 3;
}



message UpdateUserResult {
    bool success = 1;
}

message DeleteUserRequest {
    string email = 1;
}

message DeleteUserResult {
    bool success =1;
}

message ListUserRequest {

}

message ListUserResult{
    string username = 1;
    string email = 2;
    int32 phone_no = 4;

}

service Users {
  rpc CreateUser (CreateUserRequest) returns (CreateUserResult);
  rpc Check_login (CheckRequest) returns (CheckResult);
  rpc Update_User (UpdateUserRequest) returns (UpdateUserResult);
  rpc Delete_User (DeleteUserRequest) returns (DeleteUserResult);
  rpc List_User (ListUserRequest) returns (stream ListUserResult);
}


